kyo-prelude/shared/src/main/scala/kyo/kernel/Frame.scala

```diff
@@ -9,45 +9,88 @@ opaque type Frame <: AnyRef = String
 
-    case class Parsed(
-        declaringClass: String,
-        methodName: String,
-        position: Position,
-        snippetShort: String,
-        snippetLong: String
-    ) derives CanEqual:
-
-        def show: String =
-            Ansi.highlight(s"// $position $declaringClass $methodName", snippetLong, s"", position.lineNumber)
-
-        override def toString = s"Frame($declaringClass, $methodName, $position, $snippetShort)"
-    end Parsed
-
-    case class Position(
-        fileName: String,
-        lineNumber: Int,
-        columnNumber: Int
-    ) derives CanEqual:
-        override def toString = s"$fileName:$lineNumber:$columnNumber"
+    opaque type Position = String
+
+    object Position:
+        extension (self: Position)
+            def fileName: String =
+                self.substring(1, self.indexOf(':'))
+
+            def lineNumber: Int =
+                val start = self.indexOf(':') + 1
+                val end   = self.indexOf(':', start)
+                self.substring(start, end).toInt
+            end lineNumber
+
+            def columnNumber: Int =
+                val firstColon = self.indexOf(':')
+                val start      = self.indexOf(':', firstColon + 1) + 1
+                val end        = self.indexOf('|', start)
+                self.substring(start, end).toInt
+            end columnNumber
+
+            def show: String =
+                self.substring(1, self.indexOf('|'))
+        end extension
     end Position
 
-    extension (t: Frame)
-        def parse: Parsed =
-            val arr = t.split('£')
-            Parsed(
-                arr(0),
-                arr(1),
-                Position(arr(2), arr(3).toInt, arr(4).toInt),
-                arr(5).split("📍")(0).reverse.take(snippetShortMaxChars).takeWhile(_ != '\n').trim.reverse,
-                arr(5)
+    extension (self: Frame)
+        private def findNextSeparator(from: Int): Int =
+            self.indexOf('|', from)
+
+        private def parseSection(start: Int, end: Int): String =
+            if end < 0 then self.substring(start)
+            else self.substring(start, end)
+
+        def position: Position = self
+
+        def className: String =
+            val start = findNextSeparator(1) + 1
+            parseSection(start, findNextSeparator(start))
+
+        def methodName: String =
+            val firstSep = findNextSeparator(1)
+            val start    = findNextSeparator(firstSep + 1) + 1
+            parseSection(start, findNextSeparator(start))
+        end methodName
+
+        def snippet: String =
+            val firstSep  = findNextSeparator(1)
+            val secondSep = findNextSeparator(firstSep + 1)
+            val start     = findNextSeparator(secondSep + 1) + 1
+            self.substring(start)
+        end snippet
+
+        def snippetShort: String = snippet.split("📍")(0).reverse.take(snippetShortMaxChars).takeWhile(_ != '\n').trim.reverse
```

Comment:
this expensive string transformation was being done on each benchmark iteration 😱